src/AnalysisType.h

```diff
@@ -0,0 +1,420 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2019, The Souffle Developers. All rights reserved.
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AnalysisType.h
+ *
+ * A collection of analysis types that may appear in a type lattice.
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "AstType.h"
+#include "Util.h"
+#include <cassert>
+#include <sstream>
```

Comment:
should this be ostream?
Also need vector, set, and memory, and maybe some more Ast stuff?

Running iwyu would be a good idea. Something like
```
for i in *.cpp;do iwyu --Xiwyu --transitive_includes_only --std=c++11 -I. -I.. -I ~/souffle/include $i 2>>iwyuT.log;done
```

Reply:
Incidentally, don't assume iwyu is correct. Only apply what makes sense.